ResearchKit/Common/ORKPasscodeStepViewController.m

```diff
@@ -93,6 +98,26 @@ - (void)stepDidChange {
         _isPasscodeSaved = NO;
         _useTouchId = YES;
         
+        // If this has text, we should add the forgot passcode button with this title
+        if (self.forgotPasscodeText != nil && _forgotPasscodeButton == nil)
+        {
+            [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardWillShow:) name:UIKeyboardWillShowNotification object:nil];
+            
+            CGFloat x = kForgotPasscodeHorizontalPadding;
+            _originalForgotPasscodeY = self.view.bounds.size.height - kForgotPasscodeVerticalPadding - kForgotPasscodeHeight;
+            CGFloat width = self.view.bounds.size.width - 2*kForgotPasscodeHorizontalPadding;
+            UIButton* forgotPasscodeButton = [[UIButton alloc] initWithFrame:CGRectMake(x, _originalForgotPasscodeY, width, kForgotPasscodeHeight)];
+            
+            [forgotPasscodeButton setTitle:self.forgotPasscodeText forState:UIControlStateNormal];
+            [forgotPasscodeButton setTitleColor:forgotPasscodeButton.tintColor forState:UIControlStateNormal];
+            
+            [forgotPasscodeButton addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(forgotPasscodeTapped)]];
```

Comment:
Use the event rather than the tap gesture. Sometimes a user will press and hold and it's confusing if it doesn't register the event b/c it's then a "long press".